Mathlib/Probability/Moments.lean

```diff
@@ -282,6 +287,27 @@ theorem iIndepFun.cgf_sum {X : ι → Ω → ℝ}
   · rw [h_indep.mgf_sum h_meas]
   · exact (mgf_pos (h_int j hj)).ne'
 
+theorem mgf_ident_distrib (X Y : Ω → ℝ) (hident : IdentDistrib X Y μ μ) :
+  mgf X μ t = mgf Y μ t := by
+    rw [mgf, mgf]
+    apply IdentDistrib.integral_eq
+    let u := fun x => Real.exp (t * x)
+    have compX : (fun ω ↦ Real.exp (t * X ω)) = u ∘ X := rfl
+    have compY : (fun ω ↦ Real.exp (t * Y ω)) = u ∘ Y := rfl
+    rw [compX, compY]
+    exact IdentDistrib.comp hident (Measurable.exp (measurable_const_mul t))
+
+theorem mgf_sum_iid
+{X : ι → Ω → ℝ}
+(h_meas : ∀ (i : ι), Measurable (X i))
+(h_indep : ProbabilityTheory.iIndepFun (fun _ => inferInstance) X μ)
+(hident : ∀ (i j : ι), ProbabilityTheory.IdentDistrib (X i) (X j) μ μ)
+(s : Finset ι) (j : ι)
+: ∀ t : ℝ, mgf (∑ i ∈ s, X i) μ t = (mgf (X j) μ (t)) ^ s.card := by
+    intro t
+    rw [iIndepFun.mgf_sum h_indep h_meas]
+    exact Finset.prod_eq_pow_card (fun i si => mgf_ident_distrib (X i) (X j) (hident i j))
```

Comment:
Indentation is also not mathlib-like here. Let me show you how it should be:
```suggestion
theorem mgf_sum_iid
    {X : ι → Ω → ℝ}
    (h_meas : ∀ (i : ι), Measurable (X i))
    (h_indep : ProbabilityTheory.iIndepFun (fun _ => inferInstance) X μ)
    (hident : ∀ (i j : ι), ProbabilityTheory.IdentDistrib (X i) (X j) μ μ)
    (s : Finset ι) (j : ι) : 
    ∀ t : ℝ, mgf (∑ i ∈ s, X i) μ t = (mgf (X j) μ (t)) ^ s.card := by
  intro t
  rw [iIndepFun.mgf_sum h_indep h_meas]
  exact Finset.prod_eq_pow_card (fun i si => mgf_ident_distrib (X i) (X j) (hident i j))
```
(possibly regrouping some variables on common lines)